android/Dockerfile

```diff
@@ -13,6 +13,7 @@ RUN apt-get update \
         curl \
         git \
         zip \
+        build-essential\
```

Comment:
we probably need to find a way to fix the gcc version. this will download "latest" which can differ between runs of the dockerfile and cause non-reproducible builds.

Reply:
<img width="484" alt="Screenshot 2024-07-09 at 12 38 04" src="https://github.com/acrespo/apollo/assets/260114/50f564aa-9d81-49ef-9e35-283c19c2ec2d">

Reply:
We believe this is no longer needed, we introduced this to fix a compiling error before trying out the "full"/rebased changes from upstream. Those include a commit where they no longer try to compile certain files before hand. Will try removing this